ext/dep_gecode/dep_selector_to_gecode.cpp

```diff
@@ -843,7 +843,9 @@ int VersionProblem::Solve(VersionProblem * problem, VersionProblem ** solution)
         DEBUG_STREAM << problem->DebugPrefix() << "Solver Best Solution " << *solution << std::endl << std::flush;
     }
 
-    pool->Delete(*solution);
+    if (*solution) {
+        pool->Delete(*solution);
+    }
```

Comment:
This looks quite reasonable. I'm going back and forth about whether we should log something on the else side; I *think* everything that might cause this will be logged in InnerSolve, but not 100% sure.